src/modules/components/proposal/proposalActions/actions/proposalActionUpdateMetadata/proposalActionUpdateMetadata.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import { modulesCopy } from '../../../../../assets';
+import { generateProposalActionUpdateMetadata } from '../generators';
+import { type IProposalActionUpdateMetadataProps, ProposalActionUpdateMetadata } from './proposalActionUpdateMetadata';
+
+describe('<ProposalActionUpdateMetadata /> component', () => {
+    const createTestComponent = (props?: Partial<IProposalActionUpdateMetadataProps>) => {
+        const defaultProps: IProposalActionUpdateMetadataProps = {
+            action: generateProposalActionUpdateMetadata(),
+            ...props,
+        };
+
+        return <ProposalActionUpdateMetadata {...defaultProps} />;
+    };
+
+    it('renders the ToggleGroup component', () => {
+        render(createTestComponent());
+        expect(screen.getByText('Existing')).toBeInTheDocument();
+        expect(screen.getByText('Proposed')).toBeInTheDocument();
+    });
+
+    it('renders the correct terms', () => {
+        render(createTestComponent());
+        expect(screen.getByText(modulesCopy.proposalActionsUpdateMetadata.logoTerm)).toBeInTheDocument();
+        expect(screen.getByText(modulesCopy.proposalActionsUpdateMetadata.linkTerm)).toBeInTheDocument();
+        expect(screen.getByText(modulesCopy.proposalActionsUpdateMetadata.nameTerm)).toBeInTheDocument();
+        expect(screen.getByText(modulesCopy.proposalActionsUpdateMetadata.descriptionTerm)).toBeInTheDocument();
+    });
+
+    it('renders the correct existing metadata', () => {
+        const proposedMetadata = {
+            logo: 'proposed-logo.png',
+            name: 'Proposed Name',
+            description: 'Proposed DAO description',
+            links: [
+                { label: 'Proposed Link 1', href: 'https://proposed-link1.com', target: '_blank' },
+                { label: 'Proposed Link 2', href: 'https://proposed-link2.com', target: '_blank' },
+            ],
+        };
+
+        const existingMetadata = {
+            logo: 'existing-logo.png',
+            name: 'Existing Name',
+            description: 'Existing DAO description',
+            links: [
+                { label: 'Existing Link 1', href: 'https://existing-link1.com', target: '_blank' },
+                { label: 'Existing Link 2', href: 'https://existing-link2.com', target: '_blank' },
+            ],
+        };
+
+        const action = generateProposalActionUpdateMetadata({ proposedMetadata, existingMetadata });
+
+        render(createTestComponent({ action }));
+
+        expect(screen.getByText('Existing Name')).toBeInTheDocument();
+        expect(screen.getByText('Existing Link 1')).toBeInTheDocument();
+        expect(screen.getByText('Existing Link 2')).toBeInTheDocument();
+        expect(screen.getByText('Existing DAO description')).toBeInTheDocument();
+
+        fireEvent.click(screen.getByText('Proposed'));
```

Comment:
React testing-library suggests to use `userEvent` instead of `fireEvent` whenever possible
```suggestion
        userEvent.click(screen.getByText('Proposed'));
```